### ISSUE TITLE
TypeError: Class constructor EventEmitter cannot be invoked without 'new'

### ISSUE THREAD

**Issue by rodrigograca31** [twurple/twurple#414]
### Code
```ts
#!/usr/bin/env node

// PubSub Code
// ======================================

import { PubSubClient, PubSubRedemptionMessage } from "@twurple/pubsub";
import { ApiClient } from "@twurple/api";
import { StaticAuthProvider } from "@twurple/auth";

const clientId = "aaa";
const accessToken = "bbb";
const authProvider = new StaticAuthProvider(clientId, accessToken, [
	"channel:read:redemptions",
	"channel_subscriptions",
	"channel:read:hype_train",
]);
const apiClient = new ApiClient({ authProvider });

const pubSubClient = new PubSubClient();

pubSubClient.onBits("CHANEL", (message) => {
	console.log(message);
});

```

### Expected behavior

just working

### Actual Behavior

`TypeError: Class constructor EventEmitter cannot be invoked without 'new'
    at new PersistentConnection (...../node_modules/@d-fischer/connection/src/PersistentConnection.ts:36:3)
    at new BasicPubSubClient (....../node_modules/@twurple/pubsub/lib/BasicPubSubClient.js:65:28)
    at new PubSubClient (....../node_modules/@twurple/pubsub/lib/PubSubClient.js:24:88)
    `

### Environment

```
"@twurple/api": "^5.2.7",
"@twurple/auth": "^5.2.7",
"@twurple/chat": "^5.2.7",
"@twurple/pubsub": "^5.2.7",
```
node: v16.15.1

seems the issue was introduced by this change:
https://github.com/d-fischer/ircv3/commit/586b454d76633e368dffad662ad659e19fef3250
because days ago this very same code was working but now after updating the npm packages and getting the latest dependencies its broken because it now uses `@d-fischer/connection@7.0.1`

**Comment by rodrigograca31** [comment]
Actually im not sure where it was introduced because:
![image](https://user-images.githubusercontent.com/1134310/202879344-9d94844a-3a3d-474d-8e4f-83f88bf67364.png)


and I was not using `/chat` so I uninstalled it it changed to this but the issue still persists:

![image](https://user-images.githubusercontent.com/1134310/202879365-fbdfd2f2-5136-4e5f-ad0e-42b8f8176b7d.png)

**Comment by rafaelpernil2** [comment]
I did some research and this bug is introduced by a change in `ircv3` between versions 0.28.1 and 0.29.8, and by another change in `@d-fischer/typed-event-emitter` between version 3.2.2 and 3.3.0. 

The ES target version inside `tsconfig.json` has been changed to ES2019. 
![image](https://user-images.githubusercontent.com/9103141/202924630-faab52ac-918d-44fe-a8e4-8bb920229a88.png)

I have experienced this bug with `@twurple/chat`

# Differences in transpiled code

## @d-fischer/typed-event-emitter
Old version - 3.2.2
![image](https://user-images.githubusercontent.com/9103141/202921342-b7b4abfe-c6ab-45ed-acb6-aa3c472337a4.png)
New version - 3.3.0
![image](https://user-images.githubusercontent.com/9103141/202921353-2d34b4c8-122a-4615-b4a1-3b7a46311c54.png)

Changelog: https://github.com/d-fischer/typed-event-emitter/compare/v3.2.2...v3.3.0#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0R6-R7

## ircv3
Old version - 0.28.1
![image](https://user-images.githubusercontent.com/9103141/202922871-d41f08c1-aa47-4419-91d6-ba3fe37d8a3d.png)
New version - 0.29.8
![image](https://user-images.githubusercontent.com/9103141/202922785-aabd70b9-6a62-48fa-aa0d-613e2be68d35.png)

Changelog: https://github.com/d-fischer/ircv3/compare/v0.28.1...v0.29.8#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0L5-L8


In essence, if `@d-fischer/typed-event-emitter` is on version 3.3.0 in your `package-lock.json`, you also need to update `ircv3` to version 0.29.6 (at least). 
> The latest version of `ircv3` also works with `@d-fischer/typed-event-emitter` 3.2.2.

Oddly, the dependency versions are correct in Twurple but `ircv3` is not behaving correctly when updating its version.

# Solution

I found out there's a very simple fix:

1. Remove `node_modules/ircv3`
2. Run `npm install` again

Somehow the js files are not re-transpiled if the package was already installed.

# Points of improvement

I would advise NOT using "caret" versions because these bugs are very common and break functionality without any change in your source code. specially with alpha/beta software like "ircv3" that cannot follow SemVer with breaking changes. 

And also, have in mind that a new "fix" version of a dependency could be pushed with malicious code for some reason and it would be catastrophic.

**Comment by rodrigograca31** [comment]
ooohhhh..... I was removing `node_modules` and doing `npm i` but forgot to remove `package-lock.json`

doing that and then `npm i` fixed the issue! closing

**Comment by d-fischer** [comment]
TL;DR: I did some mistakes, do `npm upgrade` / `yarn upgrade` to fix them for yourself

I want to give a little bit of context here to the issue itself, but also a few of the things written in here.

## What happened

- Pushing a target change of `@d-fischer/typed-event-emitter` and `@d-fischer/connection` in a minor version and of `ircv3` in a patch version -> all of these were big mistakes, I admit that.
- Pushing a new major version of `@d-fischer/connection` to depend on at least that new version of `@d-fischer/typed-event-emitter` -> completely fine.
- Pushing a new patch version `ircv3`  to depend on that connection version -> also fine, but it also depends on `@d-fischer/typed-event-emitter`, which I did not bump.

## The issue

There could be a situation where, deep within the dependency tree, you would update `@d-fischer/typed-event-emitter`, which came with the target change, but not `ircv3`. For example, you could be updating `@twurple/eventsub`, which, like most `@twurple` packages, depends on `@d-fischer/typed-event-emitter`, but doesn't depend on `ircv3`. This would lead to "old" target code (from `ircv3`) trying to call "new" target code (from `@d-fischer/typed-event-emitter`), which leads to the error that was reported in `@twurple/chat`, as it depends on `ircv3`.

## How to fix it for yourself

```sh
yarn upgrade
# or
npm upgrade
```

This updates every package to be published as transpiled to the newer ES version targets, which fixes the error.

I pushed a fix to `ircv3` to require the newer version of `@d-fischer/typed-event-emitter` so it's much less likely to happen now - but even then, the above commands should fix it.

## Some comments & corrections

> And also, have in mind that a new "fix" version of a dependency could be pushed with malicious code for some reason and it would be catastrophic.

All packages involved are written by myself, including `ircv3`. It's extremely unlikely that I'd tamper with a dependency of my own package.

I'm pretty sure that currently, the only package in my dependency tree that is not controlled by myself is `node-fetch`, which I trust will not do anything shady, I'm happy to move away from it though as soon as global `fetch` becomes stable in node and the last version not shipping it goes end-of-life.

> I would advise NOT using "caret" versions [...] specially with alpha/beta software like "ircv3" that cannot follow SemVer with breaking changes.

`ircv3` is in the sub-1.0-but-at-least-0.1 region, so in general, you *can* differentiate between breaking and non-breaking changes, because npm understands `^0.28.1` as `>=0.28.1 <0.29`. I try to always put breaking changes in caret-incompatible versions, to follow npm's understanding of SemVer. This time it did not work out, and it's my fault. I'll try to be better, but my understanding of SemVer will stay largely the same.

Also, the reason I keep its version below 1.0 is that it's not feature complete for its namesake, not that it's not well-tested. I started writing that library **more than 5 years ago** with the intention of serving as a base for the chat client of Twurple (or, back then, Twitch.js), so it's just as battle-tested as Twurple. I wouldn't call it beta by a long shot. It just lacks some actual IRCv3 features (which Twitch does not use).

> ooohhhh..... I was removing `node_modules` and doing `npm i` but forgot to remove `package-lock.json`

You don't generally have to do things like this. `npm upgrade` does *mostly* the same thing as removing your lockfile and node_modules, just much more performant.

**Comment by rafaelpernil2** [comment]
Thanks for the detailed explanation!

I saw most dependencies were yours after writing my comment, it makes perfect sense now.

And also thank you for creating and maintaining this awesome library, it works great :)
